### PR TITLE
fix: cleanup connect

### PR DESCRIPTION
**Pull request by ronag**

### CODE REVIEW

**Review by mcollina — APPROVED** [review]
lgtm

**Comment by mcollina** [comment]
CI seems failing

File: lib/api/api-connect.js [hunk 1]
```diff
@@ -41,23 +40,8 @@ class ConnectHandler extends AsyncResource {
...
-            this.runInAsyncScope(callback, null, err, { opaque })
-          })
-        }
-        return 1
-      }
-    } else {
-      throw new SocketError('bad connect', null)
-    }
+  onHeaders () {
+    throw new SocketError('bad connect', null)
```
[the remark this answers is not in the document]
Comment (ronag) [reply lib/api/api-connect.js:44]: It shouldn't call this in case of an upgrade so something is wrong.

**Comment by KhafraDev** [comment]
CI seems to be failing on `should throw when proxy does not return 200` which was added in df4f7e0e95f5112322a96fd7a666cb28c1d48327

File: lib/api/api-connect.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (RafaelGSS) [reply lib/api/api-connect.js:44]: It should, the CONNECT request is sent to the Proxy server and the client needs to wait for a 200 status code (so headers are expected) to then an upgrade to TLS through HTTP Tunnel

[the remark this answers is not in the document]
Comment (ronag) [reply lib/api/api-connect.js:44]: I disagree. The onUpgrade method gets status.

[the remark this answers is not in the document]
Comment (ronag) [reply lib/api/api-connect.js:44]: Ok. This is fixed. Was a "bug" in client.

**Review by KhafraDev — APPROVED** [review]
lint is failing but otherwise lgtm

**Pull request merged**